### DISCUSSION TITLE
merged MAFs - changed sample name | error in oncodrive

### DISCUSSION THREAD

**Discussion by complexgenome** [PoisonAlien/maftools#1020] [Q&A] [answered]
Hi there,

I'm interested in using oncodrive function in the maftools R library.
I generated MAF files using GATK 4.4 funcotator.

I'd like to merge MAFs, but saw there's been issue with sample name:

https://github.com/PoisonAlien/maftools/discussions/822
https://github.com/PoisonAlien/maftools/issues/819

I replaced column 16th column (Tumor_Sample_Barcode) with sample name in the MAF file using bash code as:

```
grep "#" $maf_file_name > $sample_name".maf"

grep -v "#" $maf_file_name | awk -v OFS="\t" -v sample="$sample_name" 'BEGIN {FS="\t"; OFS="\t"} NR==1 {print; next}  {$(16) = sample; print}'  >>$sample_name".maf"
```
$sample_name".maf" is edited file with 16th column

I then use R to merge the MAFs:

```
maf_files = list.files(path = "/data1/users/sanjeev/drivers/create_mafs/", pattern = "*.\\.maf", full.names = TRUE)

mymaf = maftools::merge_mafs(mafs = maf_files, verbose = TRUE)
maftools::write.mafSummary(maf = mymaf, basename = "mymaf")
```

I run into error as:

> Error in data.table::rbindlist(l = maf, fill = TRUE, idcol = "Source_MAF",  : 
>   Class attribute on column 112 of item 9 does not match with column 112 of item 5.
> 

I then used help from https://github.com/PoisonAlien/maftools/issues/722
When I use it in R as:

```

required.fields = c(
  'Hugo_Symbol',
  'Chromosome',
  'Start_Position',
  'End_Position',
  'Reference_Allele',
  'Tumor_Seq_Allele2',
  'Variant_Classification',
  'Variant_Type',
  'Tumor_Sample_Barcode'
)

mafs = lapply(X = maf_files, function(x){
  data.table::fread(file = x, sep = "\t", select = required.fields, skip = "Hugo_Symbol")
})

mafs = data.table::rbindlist(l = mafs, use.names = TRUE, fill = TRUE, idcol = "Source_MAF")

oncodrive(mafs)

``` 

I run into error as:

```
oncodrive(mafs)
Error in maf@maf.silent : 
  no applicable method for `@` applied to an object of class "data.table"
In addition: Warning message:
In oncodrive(mafs) :
  Oncodrive has been superseeded by OncodriveCLUSTL. See http://bg.upf.edu/group/projects/oncodrive-clust.php

```
session info:

 R version 4.3.2 (2023-10-31)
data.table             1.15.4 
  maftools             * 2.18.0  

Let me know if any other details are needed.

**Comment by PoisonAlien** [comment, marked as answer]
Hi,

Thanks for the issue. `data.table::rbindlist()` returns a `data.table`. You must convert it to an MAF object with `read.maf()`.

```r
mafs = data.table::rbindlist(l = mafs, use.names = TRUE, fill = TRUE, idcol = "Source_MAF")
m = maftools::read.maf(mafs)
oncodrive(m)
```

Also, please take a look at the warning message. You may want to consider alternative options.
